diff --git a/drv.c b/drv.c
index 9dd7471..7fa6093 100755
--- a/drv.c
+++ b/drv.c
@@ -43,6 +43,7 @@ MODULE_LICENSE("GPL");
 #define PCI_DEVICE_ID_AMD_19H_M70H_ROOT     0x14e8
 #define PCI_DEVICE_ID_AMD_MI200_ROOT        0x14bb
 #define PCI_DEVICE_ID_AMD_MI300_ROOT        0x14f8
+#define PCI_DEVICE_ID_AMD_VANGOGH_ROOT      0x1645
 
 #define MAX_ATTRS_LEN                      13
 
@@ -451,6 +452,7 @@ static struct pci_device_id ryzen_smu_id_table[] = {
     { PCI_DEVICE(PCI_VENDOR_ID_AMD, PCI_DEVICE_ID_AMD_19H_M70H_ROOT) },
     { PCI_DEVICE(PCI_VENDOR_ID_AMD, PCI_DEVICE_ID_AMD_MI200_ROOT) },
     { PCI_DEVICE(PCI_VENDOR_ID_AMD, PCI_DEVICE_ID_AMD_MI300_ROOT) },
+    { PCI_DEVICE(PCI_VENDOR_ID_AMD, PCI_DEVICE_ID_AMD_VANGOGH_ROOT) },
     { }
 };
 MODULE_DEVICE_TABLE(pci, ryzen_smu_id_table);
diff --git a/smu.c b/smu.c
index 3c439e6..bec9dd4 100755
--- a/smu.c
+++ b/smu.c
@@ -293,6 +293,7 @@ int smu_resolve_cpu_class(struct pci_dev *dev) {
       g_smu.codename = CODENAME_MATISSE;
       break;
     case 0x90:
+    case 0x91:
       g_smu.codename = CODENAME_VANGOGH;
       break;
     default:
@@ -416,13 +417,11 @@ int smu_init(struct pci_dev *dev) {
   case CODENAME_PHOENIX:
   case CODENAME_STRIXPOINT:
   case CODENAME_HAWKPOINT:
+  case CODENAME_VANGOGH:
     g_smu.addr_rsmu_mb_cmd = 0x3B10A20;
     g_smu.addr_rsmu_mb_rsp = 0x3B10A80;
     g_smu.addr_rsmu_mb_args = 0x3B10A88;
     goto LOG_RSMU;
-  case CODENAME_VANGOGH:
-    pr_debug("RSMU Mailbox: Not supported or unknown, disabling use.");
-    goto MP1_DETECT;
   default:
     pr_err("Unknown processor codename: %d", g_smu.codename);
     return -ENODEV;
@@ -683,6 +682,7 @@ u64 smu_get_dram_base_address(struct pci_dev *dev) {
   case CODENAME_PICASSO:
   case CODENAME_RAVENRIDGE:
   case CODENAME_RAVENRIDGE2:
+  case CODENAME_VANGOGH:
     fn[0] = 0x0a;
     fn[1] = 0x3d;
     fn[2] = 0x0b;
@@ -796,6 +796,7 @@ enum smu_return_val smu_transfer_table_to_dram(struct pci_dev *dev) {
   case CODENAME_PICASSO:
   case CODENAME_RAVENRIDGE:
   case CODENAME_RAVENRIDGE2:
+  case CODENAME_VANGOGH:
     args.s.arg0 = 3;
     fn = 0x3d;
     break;
@@ -827,19 +828,10 @@ enum smu_return_val smu_transfer_2nd_table_to_dram(struct pci_dev *dev) {
   case CODENAME_PICASSO:
   case CODENAME_RAVENRIDGE:
   case CODENAME_RAVENRIDGE2:
+  case CODENAME_VANGOGH:
     args.s.arg0 = 5;
     fn = 0x3d;
     break;
-  case CODENAME_SUMMITRIDGE:
-  case CODENAME_THREADRIPPER:
-  case CODENAME_NAPLES:
-  case CODENAME_CASTLEPEAK:
-  case CODENAME_MATISSE:
-  case CODENAME_VERMEER:
-  case CODENAME_MILAN:
-  case CODENAME_CEZANNE:
-  case CODENAME_RENOIR:
-  case CODENAME_LUCIENNE:
   default:
     return SMU_Return_Unsupported;
   }
@@ -1023,6 +1015,7 @@ u32 smu_update_pmtable_size(u32 version) {
   case CODENAME_PICASSO:
   case CODENAME_RAVENRIDGE:
   case CODENAME_RAVENRIDGE2:
+  case CODENAME_VANGOGH:
     // These codenames have two PM tables, a larger (primary) one and a smaller
     // one. The size is always fixed to 0x608 and 0xA4 bytes each. Source: Ryzen
     // Master.
