diff -Nru archlinux-linux-charcoal.old/drivers/gpu/drm/amd/pm/swsmu/smu11/vangogh_ppt.c archlinux-linux-charcoal/drivers/gpu/drm/amd/pm/swsmu/smu11/vangogh_ppt.c
--- archlinux-linux-charcoal.old/drivers/gpu/drm/amd/pm/swsmu/smu11/vangogh_ppt.c	2025-11-21 19:14:02.719365804 +0100
+++ archlinux-linux-charcoal/drivers/gpu/drm/amd/pm/swsmu/smu11/vangogh_ppt.c	2025-11-21 19:18:24.635346895 +0100
@@ -2339,7 +2339,7 @@
 	if (default_power_limit)
 		*default_power_limit = ppt_limit / 1000;
 	if (max_power_limit)
-		*max_power_limit = 29;
+		*max_power_limit = 50;
 	if (min_power_limit)
 		*min_power_limit = 0;
 
